JS adapter build fails if project.version is updated to a version requiring semver translation

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

adapter/javascript

### Describe the bug

This follows up on #23512 

If `project.version` is a value that does not work with NPM, and therefore requires semver translation when updating the NPM package.json files, then maven will no longer have the correct version when looking for the JS adapter's tgz file, and an error will fail the maven build.

### Version

main

### Expected behavior

The build should run without errors

### Actual behavior

The build fails, in the install phase of the JS adapter (js/libs/keycloak-js/pom.xml):

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-install-plugin:2.5.2:install (default-install) on project keycloak-js-adapter: Failed to install artifact org.keycloak:keycloak-js-adapter:tar.gz:999.0.0.temporary-00189: /tmp/org.keycloak-keycloak-parent-24.x/js/libs/keycloak-js/target/keycloak-js-999.0.0.temporary-00189.tgz -> [Help 1]
```

### How to Reproduce?

```
./set-version.sh 999.0.0.temporary-00189
mvn clean install -DskipTests
```

### Anything else?

_No response_

--- a/js/libs/keycloak-js/pom.xml
+++ b/js/libs/keycloak-js/pom.xml
@@ -83,8 +83,8 @@
                         </goals>
                         <configuration>
                             <artifacts>
-                                <artifact>${project.basedir}/assembly.xml
-                                    <file>target/keycloak-js-${project.version}.tgz</file>
+                                <artifact>
+                                    <file>target/keycloak-js-${project.version.npm}.tgz</file>
                                     <type>tar.gz</type>
                                 </artifact>
                             </artifacts>

--- a/pom.xml
+++ b/pom.xml
@@ -35,6 +35,8 @@
     <packaging>pom</packaging>
 
     <properties>
+        <project.version.npm>999.0.0-SNAPSHOT</project.version.npm>
+
         <asciidoctor.plugin.version>1.5.8</asciidoctor.plugin.version>
 
         <jboss.repo.nexusUrl>https://s01.oss.sonatype.org/</jboss.repo.nexusUrl>

--- a/set-version.sh
+++ b/set-version.sh
@@ -11,6 +11,7 @@ fi
 
 # Maven
 mvn versions:set -DnewVersion=$NEW_VERSION -DgenerateBackupPoms=false -DgroupId=org.keycloak* -DartifactId=*
+mvn versions:set-property --non-recursive -Dproperty=project.version.npm -DnewVersion="$NEW_NPM_VERSION"
 
 # Docker
 sed -i "s/ENV KEYCLOAK_VERSION .*/ENV KEYCLOAK_VERSION $NEW_VERSION/" quarkus/container/Dockerfile
@@ -29,3 +30,6 @@ echo "$(jq '. += {"version": "'$NEW_NPM_VERSION'"}' js/libs/keycloak-js/package.
 
 # Keycloak Admin Client
 echo "$(jq '. += {"version": "'$NEW_NPM_VERSION'"}' js/libs/keycloak-admin-client/package.json)" > js/libs/keycloak-admin-client/package.json
+
+echo "New Mvn Version: $NEW_VERSION" >&2
+echo "New NPM Version: $NEW_NPM_VERSION" >&2

